### PR TITLE
fix #1547 add a SequenceFileSink

### LINKED ISSUE #1547: Add a HDFS Sink with secutiry

### PR DESCRIPTION
**Pull request by huafengw**
In this pr, I also drop the pattern that hadoop Configuration must be initiated by client side.

### CODE REVIEW

File: external/hadoopfs/src/main/scala/io/gearpump/streaming/hadoop/SequenceFileSink.scala [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+    * write message into data sink
+    * invoked at onNext() method of [[Task]]
+    * @param message wraps data to be written out
+    */
+  override def write(message: Message): Unit = {
+    val key = sequenceFormat.getKey(message)
+    val value = sequenceFormat.getValue(message)
+    writer.append(key, value)
```
Comment (manuzhang) [inline external/hadoopfs/src/main/scala/io/gearpump/streaming/hadoop/SequenceFileSink.scala]: nullable ?

File: external/hadoopfs/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# GearPump HDFS
```
Comment (manuzhang) [inline external/hadoopfs/README.md]: It seems we don't have consensus on "Gearpump" or "GearPump" yet.

File: external/hadoopfs/src/main/scala/io/gearpump/streaming/hadoop/SequenceFileSink.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  /**
+    * write message into data sink
+    * invoked at onNext() method of [[Task]]
+    * @param message wraps data to be written out
+    */
+  override def write(message: Message): Unit = {
+    val key = sequenceFormat.getKey(message)
+    val value = sequenceFormat.getValue(message)
+    if (key != null && value != null) {
+      writer.append(key, value)
```
Comment (manuzhang) [inline external/hadoopfs/src/main/scala/io/gearpump/streaming/hadoop/SequenceFileSink.scala]: sorry, I mean `writer` could be null

File: external/hadoopfs/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Gearpump HDFS
+
+Gearpump components for interacting with HDFS file systems.
+
+## Usage
+
+1. File Rotation interface
+```
```
Comment (manuzhang) [inline external/hadoopfs/README.md]: you may specify the code block as Scala such that we get syntax highlight

File: external/hbase/README.md [hunk 5]
```diff
@@ -2,20 +2,51 @@
 
 GearPump integration for [Apache HBase](https://hbase.apache.org)
 
+## Usage
+
+The message type that HBaseSink is able to handle including:
+
+ 1. Tuple4[String, String, String, String]
```
Comment (manuzhang) [inline external/hbase/README.md]: users won't understand what are in this `Tuple4`.

File: external/hbase/src/main/scala/io/gearpump/external/hbase/HBaseSink.scala [hunk 6]
```diff
@@ -28,20 +28,13 @@ import org.apache.hadoop.hbase.{TableName, HBaseConfiguration}
 import org.apache.hadoop.hbase.client.{ConnectionFactory, Put}
 import org.apache.hadoop.hbase.util.Bytes
 
-class HBaseSink(userconfig: UserConfig, tableName: String, @transient var configuration: Configuration) extends DataSink{
+class HBaseSink(userconfig: UserConfig, tableName: String) extends DataSink{
+  @transient lazy val configuration = HBaseConfiguration.create()
```
Comment (manuzhang) [inline external/hbase/src/main/scala/io/gearpump/external/hbase/HBaseSink.scala]: do we need `configuration` and `connection` elsewhere ? Why not only use them to get `table` in `open`

Comment (huafengw) [inline external/hbase/src/main/scala/io/gearpump/external/hbase/HBaseSink.scala]: connection need to be closed

**Comment by clockfly** [comment]
+1

**Pull request merged**